axum-core/src/body.rs

```diff
@@ -20,7 +20,8 @@ where
     try_downcast(body).unwrap_or_else(|body| body.map_err(Error::new).boxed_unsync())
 }
 
-pub(crate) fn try_downcast<T, K>(k: K) -> Result<T, K>
+/// Returns some mutable reference to the inner value if it is of type T, or Returns Err
+pub fn try_downcast<T, K>(k: K) -> Result<T, K>
```

Comment:
Its intentionally duplicated. We don't want `try_downcast` to be part of axum-core's public API.

Reply:
I'm not quite sure whether really need to fix it.
I found the previous commit and issue about the origin of axum-core,it is used as the foundation for the rest of axum crates.
As a developer, Duplicated Code is a code smell in my experience. Because when you want to change them, multiple codes need to be changed. 
there are some common items (eg: try_downcast, macro_rules! composite_rejection) in both axum and axum-core.
Are there plans to refactor Duplicated Code in the future?  Or what do you think about this

Reply:
Duplicated code isn't always bad. Its a trade-off. In this case we choose a little bit of duplication over adding these to the public API.